docs/reference/mapping.asciidoc

```diff
@@ -5,14 +5,39 @@
 --
 
 Mapping is the process of defining how a document, and the fields it contains,
-are stored and indexed. For instance, use mappings to define:
-
-* which string fields should be treated as full text fields.
-* which fields contain numbers, dates, or geolocations.
-* the <<mapping-date-format,format>> of date values.
-* custom rules to control the mapping for
+are stored and indexed. You can choose how and where to map your data. Each
+method provides different benefits based on where you are in your data journey:
+
+* <<dynamic-field-mapping,Dynamic field mappings>> allow you to experiment with
+and explore data when you’re just getting started. {es} will add new fields
+automatically, just by indexing a document. You can add fields to the top-level
+mapping, and to inner <<object,`object`>>  and <<nested,`nested`>> fields.
+* <<runtime-search-request,Runtime fields in search requests>> are flexible,
+allowing you to experiment with different mapping options.
+* <<runtime-mapping-fields,Runtime fields in index mappings>> allow you to make
+schema changes without reindexing. You can use these fields in conjunction with
+indexed fields to balance resource usage and performance. Your index will be
+smaller, but with slower search performance.
+* <<explicit-mapping,Explicit mappings>> provide the best search performance
```

Comment:
I am trying to understand how explicit mappings would require more storage space compared to dynamic field mappings. It depends I would say, and also around search performance, it may depend.

Reply:
Yes, this is tricky! With dynamic mapping and explicit mapping (with the exception of runtime fields), you're indexing fields. So, either method would increase the index size. I'm trying to separate runtime fields from [explicit mapping](https://elasticsearch_66530.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/explicit-mapping.html), though one form of runtime fields is still explicit mapping.

Maybe this statement should be more like:
>Explicit mappings typically provide the best search performance. However, you must reindex your data to modify the mappings, which can be costly.